File: Basics/While-Loops/Exercise/Resources/vacation.py
needed_money = float(input())
current_money = float(input())
total_days_counter = 0
spend_days_counter = 0
is_spend_five = False

while current_money < needed_money:
    total_days_counter += 1
    action_kind = input()
    action_money = float(input())

    if action_kind == "spend":
        if action_money > current_money:
            action_money = current_money

        spend_days_counter += 1
        current_money -= action_money

    elif action_kind == "save":
        spend_days_counter = 0
        current_money += action_money

    if spend_days_counter >= 5:
        is_spend_five = True
        break

if is_spend_five:
    print(f"You can't save the money.\n"
          f"{total_days_counter}")
else:
    print(f"You saved the money for {total_days_counter} days.")

File: JS-Basics/0-5-While-Loops/Exercises/vacation.js
function isVacationPossible(input) {
    let neededMoney = Number(input.shift());
    let currentMoney = Number(input.shift())
    let totalDays = 0; let spendDays = 0; let i = 0;
    while (currentMoney < neededMoney) {
        let command = input[i]; let money = Number(input[i + 1]);
        if (command == "spend") {
            if (money > currentMoney) {money = currentMoney;}
            currentMoney -= money;
            spendDays++; totalDays++; i += 2;
            if (spendDays == 5 || i == input.length) {
                return `You can't save the money.\n${totalDays}`;
            }
            continue;
        }
        spendDays = 0;
        currentMoney += money; i += 2; totalDays++;
    }
    return `You saved the money for ${totalDays} days.`;
}


console.log(isVacationPossible(["2000",
"1000",
"spend",
"1200",
"save",
"2000"]));
console.log(isVacationPossible(["110",
"60",
"spend",
"10",
"spend",
"10",
"spend",
"10",
"spend",
"10",
"spend",
"10"]));